src/Futhark/Optimise/ArrayShortCircuiting/LastUse.hs

```diff
@@ -0,0 +1,347 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- | Playground for work on merging memory blocks
+--
+-- Last-Use analysis of a Futhark program in aliased explicit-memory lore form.
+-- Takes as input such a program or a function and produces a 'M.Map VName
+-- Names', in which the key identified the let stmt, and the list argument
+-- identifies the variables that were lastly used in that stmt.  Note that the
+-- results of a body do not have a last use, and neither do a function
+-- parameters if it happens to not be used inside function's body.  Such cases
+-- are supposed to be treated separately.
+--
+-- This pass is different from 'Futhark.Analysis.LastUse' in that memory blocks
```

Comment:
Haddock needs double quotes to link modules.

Reply:
Weird.